src/pages/Facility/locations/Utils.tsx

```diff
@@ -0,0 +1,13 @@
+import { LocationList as LocationListType } from "@/types/location/location";
+
+export function getParentChain(location: LocationListType): Set<string> {
+  const parentIds = new Set<string>();
+  let current = location.parent;
+
+  while (current) {
+    parentIds.add(current.id);
+    current = current.parent;
+  }
+
+  return parentIds;
+}
```

Comment:
let's return array of location objects instead..., let the consumer of this function decide whether they want `id` only or not. some use cases might need along with names as well.